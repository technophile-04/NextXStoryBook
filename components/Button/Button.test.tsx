import React from "react";
import { render, screen } from "@testing-library/react";
import { Button } from "./Button";
import userEvent from "@testing-library/user-event";

describe("Button test case", () => {
  it("Render check", () => {
    const onClick = jest.fn();
    const { asFragment } = render(
      <Button onClick={onClick}>This is button</Button>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("Should check onClick callback", () => {
    const onClick = jest.fn();
    render(<Button onClick={onClick}>Button</Button>);
    const element = screen.getByRole("button");
    userEvent.click(element);

    expect(onClick).toHaveBeenCalled();
  });
});
