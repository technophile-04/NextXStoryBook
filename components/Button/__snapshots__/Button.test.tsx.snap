// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test case Render check 1`] = `
<DocumentFragment>
  <button
    class="css-o0uw0a"
    color="primary"
  >
    This is button
  </button>
</DocumentFragment>
`;
